crates/cxx-qt-gen/src/generator/cpp/qenum.rs

```diff
@@ -0,0 +1,111 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use indoc::formatdoc;
+use syn::Result;
+
+use crate::{
+    generator::utils::cpp::Indent,
+    parser::{mappings::ParsedCxxMappings, qenum::ParsedQEnum},
+};
+
+use super::qobject::GeneratedCppQObjectBlocks;
+
+pub fn generate(
+    qenums: &[ParsedQEnum],
+    cxx_mappings: &ParsedCxxMappings,
+) -> Result<GeneratedCppQObjectBlocks> {
+    let mut generated = GeneratedCppQObjectBlocks::default();
+
+    for qenum in qenums {
+        let enum_name = &qenum.ident.to_string();
+
+        let mut qualified_name = cxx_mappings.cxx(enum_name);
+        // TODO: this is a workaround for cxx_mappings.cxx not always returning a fully-qualified
+        // identifier.
+        // Once https://github.com/KDAB/cxx-qt/issues/619 is fixed, this can be removed.
```

Comment:
Right, one idea i had was to make the `cxx` method always add a `::` or to add every type even if it doesn't have a namespace to the namespaces mappings.

But likely something to improve with the tree structure again too.

Reply:
Right, I've actually tried doing that, but that generated stuff like: `:: ::std::unique_ptr`, so I think we should do this properly in a follow-up commit.

Reply:
Right, separate commit and tracked with that issue :-)